dds/DCPS/TopicImpl.cpp

```diff
@@ -35,9 +34,10 @@ TopicImpl::TopicImpl(const RepoId                   topic_id,
     qos_(qos),
     listener_mask_(mask),
     listener_(DDS::TopicListener::_duplicate(a_listener)),
-    id_(topic_id),
+    id_(GUID_UNKNOWN),
     monitor_(0)
 {
+  last_inconsistent_topic_count_ = 0;
```

Comment:
Tracking the "last" count as member data doesn't seem to be necessary based on how this pattern works in other parts of the code.  I also don't think it's computing total_count_change correctly for cases that don't have a listener.  Please see other `total_count_change`s and interactions with the `get_*_status` method, like the ones in DataWriterImpl.